### PR TITLE
Add Kernel driver patches for CIG CS6436-56P device

### PR DESCRIPTION
**Pull request by michealylj1**
Add Kernel driver patches for CIG CS6436-56P device：
ASIC Vendor: Nephos
Switch ASIC: NP8366

Signed-off-by: yanglijie@cigtech.com

### CODE REVIEW

File: patch/config-cig-cs6436-serail.patch [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ # CONFIG_INTEL_SOC_PMIC is not set
+-CONFIG_MFD_INTEL_LPSS=m
+-CONFIG_MFD_INTEL_LPSS_ACPI=m
+-CONFIG_MFD_INTEL_LPSS_PCI=m
++CONFIG_MFD_INTEL_LPSS=y
++CONFIG_MFD_INTEL_LPSS_ACPI=y
++CONFIG_MFD_INTEL_LPSS_PCI=y
+ # CONFIG_MFD_JANZ_CMODIO is not set
+-CONFIG_MFD_KEMPLD=m
++CONFIG_MFD_KEMPLD=y
```
Comment (lguohan) [inline patch/config-cig-cs6436-serail.patch]: why change to kernel built-in module?

File: patch/config-cig-cs6436-serail.patch [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ # CONFIG_MFD_88PM800 is not set
+ # CONFIG_MFD_88PM805 is not set
+ # CONFIG_MFD_88PM860X is not set
+@@ -6130,9 +6130,9 @@ CONFIG_INTEL_IOATDMA=m
+ CONFIG_INTEL_MIC_X100_DMA=m
+ # CONFIG_QCOM_HIDMA_MGMT is not set
+ # CONFIG_QCOM_HIDMA is not set
+-CONFIG_DW_DMAC_CORE=m
++CONFIG_DW_DMAC_CORE=y
```
Comment (lguohan) [inline patch/config-cig-cs6436-serail.patch]: why change to kernel built-in module?

File: patch/config-cig-cs6436-serail.patch [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ CONFIG_DW_DMAC=m
+-# CONFIG_DW_DMAC_PCI is not set
++CONFIG_DW_DMAC_PCI=y
```
Comment (lguohan) [inline patch/config-cig-cs6436-serail.patch]: please keep it as a module

File: patch/config-cig-cs6436-serail.patch [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+@@ -4169,11 +4169,11 @@ CONFIG_MFD_CORE=m
+ CONFIG_LPC_ICH=m
+ CONFIG_LPC_SCH=m
+ # CONFIG_INTEL_SOC_PMIC is not set
+-CONFIG_MFD_INTEL_LPSS=m
+-CONFIG_MFD_INTEL_LPSS_ACPI=m
+-CONFIG_MFD_INTEL_LPSS_PCI=m
++CONFIG_MFD_INTEL_LPSS=y
++CONFIG_MFD_INTEL_LPSS_ACPI=y
++CONFIG_MFD_INTEL_LPSS_PCI=y
```
Comment (lguohan) [inline patch/config-cig-cs6436-serail.patch]: please do not change to built-in module.

File: patch/driver-lpss-ignore-1st-i2c-bus.patch [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ 	{ PCI_VDEVICE(INTEL, 0x5ab0), (kernel_ulong_t)&apl_i2c_info },
+ 	{ PCI_VDEVICE(INTEL, 0x5ab2), (kernel_ulong_t)&apl_i2c_info },
+@@ -166,8 +166,8 @@
+ 	{ PCI_VDEVICE(INTEL, 0x5ab6), (kernel_ulong_t)&apl_i2c_info },
+ 	{ PCI_VDEVICE(INTEL, 0x5ab8), (kernel_ulong_t)&apl_i2c_info },
+ 	{ PCI_VDEVICE(INTEL, 0x5aba), (kernel_ulong_t)&apl_i2c_info },
+-	{ PCI_VDEVICE(INTEL, 0x5abc), (kernel_ulong_t)&bxt_uart_info },
+-	{ PCI_VDEVICE(INTEL, 0x5abe), (kernel_ulong_t)&bxt_uart_info },
++	//{ PCI_VDEVICE(INTEL, 0x5abc), (kernel_ulong_t)&bxt_uart_info },
++	//{ PCI_VDEVICE(INTEL, 0x5abe), (kernel_ulong_t)&bxt_uart_info },
```
Comment (lguohan) [inline patch/driver-lpss-ignore-1st-i2c-bus.patch]: please explain the reason to comment them out.

File: patch/config-cig-cs6436-serail.patch (hunk 4 above)
Comment (michealylj1) [reply patch/config-cig-cs6436-serail.patch]: I will change it to modules and then install it during the system initialization.

File: patch/config-cig-cs6436-serail.patch (hunk 1 above)
Comment (michealylj1) [reply patch/config-cig-cs6436-serail.patch]: I don't modify it explicitly. It's updated after I run the command "make menuconfig". I will update it soon.

**Pull request closed without merging**